fontforgeexe/bitmapview.c

```diff
@@ -761,23 +761,23 @@ static void BVExpose(BitmapView *bv, GWindow pixmap, GEvent *event ) {
 	Color col = (view_bgcol<0x808080)
 		? (bv->bc->byte_data ? 0x008800 : 0x004400 )
 		: 0x00ff00;
-	memset(&cvtemp,'\0',sizeof(cvtemp));
-	cvtemp.v = bv->v;
-	cvtemp.width = bv->width;
-	cvtemp.height = bv->height;
-	cvtemp.scale = bv->scscale*bv->scale;
-	cvtemp.xoff = bv->xoff/* *bv->scscale*/;
-	cvtemp.yoff = bv->yoff/* *bv->scscale*/;
-	cvtemp.b.sc = bv->bc->sc;
-	cvtemp.b.drawmode = dm_fore;
-
-	clip.width = event->u.expose.rect.width/cvtemp.scale;
-	clip.height = event->u.expose.rect.height/cvtemp.scale;
-	clip.x = (event->u.expose.rect.x-cvtemp.xoff)/cvtemp.scale;
-	clip.y = (cvtemp.height-event->u.expose.rect.y-event->u.expose.rect.height-cvtemp.yoff)/cvtemp.scale;
-	CVDrawSplineSet(&cvtemp,pixmap,cvtemp.b.sc->layers[ly_fore].splines,col,false,&clip);
-	for ( refs = cvtemp.b.sc->layers[ly_fore].refs; refs!=NULL; refs = refs->next )
-	    CVDrawSplineSet(&cvtemp,pixmap,refs->layers[0].splines,col,false,&clip);
+	cvtemp = CharViewCreateExtended(bv->bc->sc, bv->fv, BVCurEnc(bv), 0);
```

Comment:
Why is this now needed? Because cvtabs isn't initialised? Also if we do this, we're now missing a free somewhere

Reply:
Yes that's why. If I don't do this I get a segfault. If you don't set `show` to 0 you crash Xorg (at least on my system...), but doing it this way seems fine to me. I'll add the free.

Reply:
Hmm, sounds like a bit of a code smell tbh. Do you have a backtrace?

Reply:
Sorry, I learned the hard way that Xorg would crash so I don't.

Reply:
I need to go to work soon, but I've done some further checking, and this solution isn't correct.  I didn't realise it earlier, but you're creating and destroying a new charview in the expose handler. This gets very expensive, you can see this if you try to interact/draw in that view.

So one thing was that tabbed feature was only ever implemented in the outline view and not the bitmap view, which is why this is breaking. Adding tabbed support to the bitmap view is probably going to be more work than just this, so you might be better off trying to figure out if there's a way to fix this without adding that support.